### DISCUSSION TITLE
About paragraph recognition

### DISCUSSION THREAD

**Discussion by jyyang621** [jsvine/pdfplumber#1112] [Q&A]
Please describe, in as much detail as possible, your proposal and how it would improve your experience with pdfplumber.

If I want to know the strategy in pdfplumber about paragraph recognition, which part of code should I look at?

**Comment by jsvine** [comment]
Hi @jyyang621, and thanks for your interest in `pdfplumber`. Generally speaking, `pdfplumber` does not perform **paragraph** recognition, but rather word and text-line recognition. The main part of the text-extraction code can be found in https://github.com/jsvine/pdfplumber/blob/stable/pdfplumber/utils/text.py
